Gemfile

```diff
@@ -1,10 +1,12 @@
+# frozen_string_literal: true
+
 source 'https://rubygems.org'
 
 gemspec
 
-gem 'cucumber'
+gem 'cucumber', '~> 2.4.0'
```

Comment:
Maybe these should actually be moved to the gemspec file?  I think the best practice currently is to have them in the gemspec - although that does seem to have been a point of contention for some time.

Admittedly 8 years old:
http://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/

Reply:
I'm going to push a lot of this work into additional TODO's for several reasons
1) It's the best thing to try to tackle items in isolation
2) Some of these have been long-standing. And I'm still keen on only changing the bits I'm fully comfortable with
3) Whilst travis is being reworked, I'm a bit apprehensive on changing anything structural

Hope that's ok